File: bindings/java-jni/src/main/java/org/maplibre/nativejni/runtime/OfflineOperationResultKind.java
package org.maplibre.nativejni.runtime;

/** Offline database operation result kind reported by completion events. */
public enum OfflineOperationResultKind {
  NONE(0),
  REGION(1),
  OPTIONAL_REGION(2),
  REGION_LIST(3),
  REGION_STATUS(4),
  UNKNOWN(-1);

  private final int nativeValue;

  OfflineOperationResultKind(int nativeValue) {
    this.nativeValue = nativeValue;
  }

  public int nativeValue() {
    return nativeValue;
  }

  public static OfflineOperationResultKind fromNative(int nativeValue) {
    for (var kind : values()) {
      if (kind.nativeValue == nativeValue) {
        return kind;
      }
    }
    return UNKNOWN;
  }
}

File: bindings/kotlin-native/src/nativeMain/kotlin/org/maplibre/nativeffi/runtime/OfflineOperationResultKind.kt
package org.maplibre.nativeffi.runtime

import kotlin.jvm.JvmInline

/** Offline operation result kind reported by completion events. */
@JvmInline
public value class OfflineOperationResultKind(public val nativeValue: Int) {
  public companion object {
    public val NONE: OfflineOperationResultKind = OfflineOperationResultKind(0)
    public val REGION: OfflineOperationResultKind = OfflineOperationResultKind(1)
    public val OPTIONAL_REGION: OfflineOperationResultKind = OfflineOperationResultKind(2)
    public val REGION_LIST: OfflineOperationResultKind = OfflineOperationResultKind(3)
    public val REGION_STATUS: OfflineOperationResultKind = OfflineOperationResultKind(4)

    internal fun fromNative(nativeValue: UInt): OfflineOperationResultKind =
      fromNative(nativeValue.toInt())

    internal fun fromNative(nativeValue: Int): OfflineOperationResultKind =
      OfflineOperationResultKind(nativeValue)
  }
}
